client/p.go

```diff
@@ -97,98 +102,237 @@ type p struct {
 	checker internal_platformvm.Checker
 }
 
-func (pc *p) Client() platformvm.Client            { return pc.cli }
-func (pc *p) Checker() internal_platformvm.Checker { return pc.checker }
-
-func (pc *p) Balance(key key.Key) (uint64, error) {
-	pb, err := pc.cli.GetBalance(key.P())
+func ledgerAddress(l *ledger.Ledger) (string, ids.ShortID) {
+	_, pk, err := l.Address("fuji", 0, 0)
 	if err != nil {
-		return 0, err
+		panic(err)
 	}
-	return uint64(pb.Balance), nil
+	fullAddr, err := formatting.FormatAddress("P", "fuji", pk[:])
+	if err != nil {
+		panic(err)
+	}
+	return fullAddr, pk
 }
 
-// ref. "platformvm.VM.newCreateSubnetTx".
-func (pc *p) CreateSubnet(
-	ctx context.Context,
-	k key.Key,
-	opts ...OpOption,
-) (subnetID ids.ID, took time.Duration, err error) {
-	ret := &Op{}
-	ret.applyOpts(opts)
+// Sign this transaction with the provided ledger
+func ledgerSignTx(tx *platformvm.Tx, inputs int, c codec.Manager, device *ledger.Ledger) error {
+	unsignedBytes, err := c.Marshal(platformvm.CodecVersion, &tx.UnsignedTx)
+	if err != nil {
+		return fmt.Errorf("couldn't marshal UnsignedTx: %w", err)
+	}
 
-	fi, err := pc.info.GetTxFee()
+	// Attach credentials
+	hash := hashing.ComputeHash256(unsignedBytes)
+	cred := &secp256k1fx.Credential{
+		Sigs: make([][crypto.SECP256K1RSigLen]byte, 1),
+	}
+	sig, err := device.SignHash(hash, [][]uint32{{0, 0}})
 	if err != nil {
-		return ids.Empty, 0, err
+		return fmt.Errorf("problem generating credential: %w", err)
+	}
+	copy(cred.Sigs[0][:], sig[0])
+	// CLEANUP
+	for i := 0; i < inputs; i++ {
+		tx.Creds = append(tx.Creds, cred) // Attach credential
 	}
-	createSubnetTxFee := uint64(fi.CreateSubnetTxFee)
 
-	zap.L().Info("creating subnet",
-		zap.Bool("dryMode", ret.dryMode),
-		zap.String("assetId", pc.assetID.String()),
-		zap.Uint64("createSubnetTxFee", createSubnetTxFee),
-	)
-	ins, returnedOuts, _, signers, err := pc.stake(k, createSubnetTxFee)
+	signedBytes, err := c.Marshal(platformvm.CodecVersion, tx)
 	if err != nil {
-		return ids.Empty, 0, err
+		return fmt.Errorf("couldn't marshal ProposalTx: %w", err)
 	}
+	tx.Initialize(unsignedBytes, signedBytes)
+	return nil
+}
 
-	utx := &platformvm.UnsignedCreateSubnetTx{
-		BaseTx: platformvm.BaseTx{BaseTx: avax.BaseTx{
-			NetworkID:    pc.networkID,
-			BlockchainID: pc.pChainID,
-			Ins:          ins,
-			Outs:         returnedOuts,
-		}},
-		Owner: &secp256k1fx.OutputOwners{
-			// [threshold] of [ownerAddrs] needed to manage this subnet
-			Threshold: 1,
+func ledgerManagerSpends(addr ids.ShortID, outputs []*avax.UTXO, opts ...key.OpOption) (
+	totalBalanceToSpend uint64,
+	inputs []*avax.TransferableInput,
+) {
```

Comment:
No signers returned? Seems like a different `interface`?

Reply:
Root of the issue: 
<img width="1545" alt="image" src="https://user-images.githubusercontent.com/13023275/156979314-52a06780-a7f0-4862-a859-7524c97d8214.png">